src/excalidraw-app/index.tsx

```diff
@@ -283,37 +274,34 @@ function ExcalidrawApp(props: { collab: CollabContext }) {
   };
 
   return (
-    <Excalidraw
-      ref={excalidrawRef}
-      onChange={onChange}
-      width={dimensions.width}
-      height={dimensions.height}
-      initialData={initialStatePromiseRef.current}
-      user={{ name: collab.username }}
-      onCollabButtonClick={collab.onCollabButtonClick}
-      isCollaborating={collab.isCollaborating}
-      onPointerUpdate={collab.onPointerUpdate}
-    />
+    <TopErrorBoundary>
+      <Excalidraw
+        ref={excalidrawRef}
+        onChange={onChange}
+        width={dimensions.width}
+        height={dimensions.height}
+        initialData={initialStatePromiseRef.current}
+        user={{ name: collab.username }}
+        onCollabButtonClick={collab.onCollabButtonClick}
+        isCollaborating={collab.isCollaborating}
+        onPointerUpdate={collab.onPointerUpdate}
+      />
+    </TopErrorBoundary>
```

Comment:
TopErrorBoundary should live above CollabWrapper so it can catch its error, right?

Reply:
good catch!. yep, but wrapping TopErrorBoundary in AppWithCollab doesn't seem right as its responsibility should be to wrap collab only. 
so we can either wrap in`src/index.tsx` or export `withErrorBoundary` which wraps the App with `TopErrorBoundary` from this file. WDYT?

Reply:
If it's just a naming issue, what about https://github.com/excalidraw/excalidraw/pull/2313/commits/f97e3236253db10887cb45e3338f8dfb16db7074

It also simplifies the code by removing the HOC abstraction.

Reply:
Yep looks good, i have named it so in dev tools it doesn't show as anonymous https://github.com/excalidraw/excalidraw/pull/2313/commits/8c7a5bd60bd24ee52cff1334220fc0687d947e6e